cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -0,0 +1,94 @@
+# Various Ruby helper methods to generate Varnish VCL from the base configuration.
+
+# Basic regex matcher for the query part of a URL.
+QUERY_REGEX = "(\\?[a-z0-9]+)?$"
+
+def path_to_regex(path)
+  path = "^/#{path.sub(/^\//,'')}"
+  if path[-1] == '*'
+    path.slice!(-1)
+  else
+    path = path + QUERY_REGEX
+  end
+  path
+end
+
+# Returns a regex-matcher string based on an array of filename extensions.
+def extensions_to_regex(exts)
+  return [] if exts.empty?
+  ["(?i)(#{exts.join('|').gsub('.','\.')})#{QUERY_REGEX}"]
+end
+
+# Returns a regex-conditional string fragment based on the provided behavior.
+# if backend=true, call `bereq` instead of `req`.
+# if proxy=true, ignore extension-based behaviors (e.g., *.png).
+def paths_to_regex(behavior, backend=false, proxy=false)
+  req = backend ? 'bereq' : 'req'
```

Comment:
The only part of the behavior access is the path;  how about just taking the path as a parameter?  (This will make the documentation more specifici and make this easier to test.)